pypsa/component_attrs/generators.csv

```diff
@@ -2,37 +2,49 @@ attribute,type,unit,default,description,status
-min_up_time,int,snapshots,0,"Minimum number of snapshots for status to be 1.  Only read if committable is True.",Input (optional)
-min_down_time,int,snapshots,0,"Minimum number of snapshots for status to be 0.  Only read if committable is True.",Input (optional)
-up_time_before,int,snapshots,1,"Number of snapshots that the generator was online before network.snapshots start. Only read if committable is True and min_up_time is non-zero.",Input (optional)
-down_time_before,int,snapshots,0,"Number of snapshots that the generator was online before network.snapshots start. Only read if committable is True and min_down_time is non-zero.",Input (optional)
+start_up_cost,float,currency,0,Cost to start up the generator. Only read if committable is True.,Input (optional)
+shut_down_cost,float,currency,0,Cost to shut down the generator. Only read if committable is True.,Input (optional)
+min_up_time,int,snapshots,0,Minimum number of snapshots for status to be 1.  Only read if committable is True.,Input (optional)
+min_down_time,int,snapshots,0,Minimum number of snapshots for status to be 0.  Only read if committable is True.,Input (optional)
+up_time_before,int,snapshots,1,Number of snapshots that the generator was online before network.snapshots start. Only read if committable is True and min_up_time is non-zero.,Input (optional)
+down_time_before,int,snapshots,0,Number of snapshots that the generator was online before network.snapshots start. Only read if committable is True and min_down_time is non-zero.,Input (optional)
 ramp_limit_up,float,per unit,NaN,"Maximum active power increase from one snapshot to the next, per unit of the nominal power. Ignored if NaN.",Input (optional)
 ramp_limit_down,float,per unit,NaN,"Maximum active power decrease from one snapshot to the next, per unit of the nominal power. Ignored if NaN.",Input (optional)
-ramp_limit_start_up,float,per unit,1.,"Maximum active power increase at start up, per unit of the nominal power.   Only read if committable is True.",Input (optional)
-ramp_limit_shut_down,float,per unit,1.,"Maximum active power decrease at shut down, per unit of the nominal power.   Only read if committable is True.",Input (optional)
-p,series,MW,0.,active power at bus (positive if net generation),Output
-q,series,MVar,0.,reactive power (positive if net generation),Output
-p_nom_opt,float,MW,0.,Optimised nominal power.,Output
+ramp_limit_start_up,float,per unit,1,"Maximum active power increase at start up, per unit of the nominal power.   Only read if committable is True.",Input (optional)
+ramp_limit_shut_down,float,per unit,1,"Maximum active power decrease at shut down, per unit of the nominal power.   Only read if committable is True.",Input (optional)
+type_of_control_strategy,string,n/a,n/a,"Inverter control type, supported types are: q_v (v1, v2, v3, v4, s_nom, power_factor, damper), fixed_cosphi (power_factor) , and cosphi_p (set_p1,  set_p2, s_nom, power_factor_min, p_ref) .",Input (optional)
```

Comment:
I could imagine renaming `type_of_control_strategy` to `control_strategy` throughout for brevity.